### PR TITLE
Redis

### LINKED ISSUE #21: [Parser]: Redis
Now that the core functionality has been re-written, it would be great to actually include Redis in the request logging. 

Looking for: package under the name `@ogma/platform-redis` that exports a class `RedisInterceptorParser` that extends `AbstractInterceptorService` from the `@ogma/nestjs-module` package.

The output of this is still to be determined. Some values may need to be hard-coded

### PR DESCRIPTION
**Pull request by jeffminsungkim**
Definitely, there are things to be checked.

### CODE REVIEW

File: packages/nestjs-module/src/interceptor/providers/redis-interceptor.service.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { Injectable } from '@nestjs/common';
+import { AbstractInterceptorService } from './abstract-interceptor.service';
+
+@Injectable()
+export abstract class RedisInterceptorService extends AbstractInterceptorService {}
```
Comment (jmcdo29) [inline packages/nestjs-module/src/interceptor/providers/redis-interceptor.service.ts]: Was there a reason for the new class the `nestjs-module` package?

File: packages/platform-redis/src/redis-interceptor.service.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { ExecutionContext, Injectable } from '@nestjs/common';
+import { PATTERN_METADATA } from '@nestjs/microservices/constants';
+import { AbstractInterceptorService } from '@ogma/nestjs-module';
+
+@Injectable()
+export class RedisParser extends AbstractInterceptorService {
+  getCallPoint(context: ExecutionContext): any {
+    return this.reflector.get(PATTERN_METADATA, context.getHandler()).rpc;
```
Comment (jmcdo29) [inline packages/platform-redis/src/redis-interceptor.service.ts]: What's the reason for adding `.rpc`? [I think I've handled this differently](https://github.com/jmcdo29/ogma/blob/master/packages/platform-tcp/src/tcp-interceptor.service.ts#L8) in the TCP parser

**Review by jmcdo29 — CHANGES_REQUESTED** [review]
Overall looking good. I want to get this added to the integration test as well. `./integration/tests/rpc.spec.ts` should get you on the right track. It should be as easy as adding in the parser and options for a redis microservice, as well as updating the `docker-compose.yml` to start up Redis. Let me know if you'd like some help or if you've got it managed. Thanks again for the help!

File: packages/nestjs-module/src/interceptor/providers/redis-interceptor.service.ts (hunk 1 above)
Comment (jeffminsungkim) [reply packages/nestjs-module/src/interceptor/providers/redis-interceptor.service.ts]: Sorry, I've removed it.

File: packages/platform-redis/src/redis-interceptor.service.ts (hunk 2 above)
Comment (jeffminsungkim) [reply packages/platform-redis/src/redis-interceptor.service.ts]: Good catch! 😅

**Comment by jeffminsungkim** [comment]
I've added an integration test.

File: packages/platform-redis/src/redis-interceptor.service.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+export class RedisParser extends AbstractInterceptorService {
+  getCallPoint(context: ExecutionContext): string {
+    return JSON.stringify(
+      this.reflector.get(PATTERN_METADATA, context.getHandler()),
+    );
+  }
+
+  getCallerIp(context: ExecutionContext): any {
+    const data = this.getData(context);
+    return data?.ip || '';
```
Comment (jeffminsungkim) [inline packages/platform-redis/src/redis-interceptor.service.ts]: This line did not cover.

```typescript
describe('getCallerIp', () => {
  it('should return an empty string', () => {
    const ctxMock = createMock<ExecutionContext>({
      switchToRpc: () => ({
        getData: () => ({}),
      }),
    });
    expect(parser.getCallerIp(ctxMock)).toBe('');
  });
  it('should return the ip', () => {
    const ctxMock = createMock<ExecutionContext>({
      switchToRpc: () => ({
        getData: () => ({
          ip: '127.0.0.1',
        }),
      }),
    });
    expect(parser.getCallerIp(ctxMock)).toBe('127.0.0.1');
  });
});
```
Any ideas?

Comment (jmcdo29) [reply packages/platform-redis/src/redis-interceptor.service.ts]: Uh, yeah, you can add in one more test like so:
```ts
it('should return an empty string', () => {
  const ctxMock = createMock<ExecutionContext>({
    switchToRpc: () => ({
      getData: () => undefined,
    })
  });
  expect(parser.getCallerIp(ctxMock)).toBe('');
});
```
For now, it's fine without this, it's a case that would practically never happen.

Comment (jeffminsungkim) [reply packages/platform-redis/src/redis-interceptor.service.ts]: Should I add one?

Comment (jmcdo29) [reply packages/platform-redis/src/redis-interceptor.service.ts]: Your call. I'm not too worried about it

**Review by jmcdo29 — APPROVED** [review]

**Pull request merged**